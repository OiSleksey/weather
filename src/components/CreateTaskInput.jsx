import React, { Component } from 'react';
import PropTypes from 'prop-types';

class CreateTaskInput extends Component {
  state = {
    value: '',
  };

  handleChange = event => {
    this.setState({
      value: event.target.value,
    });
  };

  handleClick = () => {
    this.props.sendInputData(this.state);
    this.setState({
      value: '',
    });
  };

  render() {
    return (
      <div className="create-task">
        <input
          type="text"
          className="create-task__input"
          value={this.state.value}
          onChange={this.handleChange}
        />
        <button className="btn" onClick={this.handleClick}>
          Create
        </button>
      </div>
    );
  }
}

export default CreateTaskInput;

CreateTaskInput.propTypes = {
  sendInputData: PropTypes.func.isRequired,
};
