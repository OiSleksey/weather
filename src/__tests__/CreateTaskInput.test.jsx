import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import CreateTaskInput from '../CreateTaskInput';
import { createSerializer } from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

configure({ adapter: new Adapter() });

describe('<CreateTaskInput.>', () => {
  it('should create task if input "text" and click on submit', () => {
    const mockHandleChange = jest.fn();
    const wrapperComponent = shallow(
      <CreateTaskInput sendInputData={mockHandleChange} />
    );
    const fakeEvent = {
      target: { value: 'Task number 1' },
    };
    wrapperComponent.find('.create-task__input').simulate('change', fakeEvent);
    wrapperComponent.find('.btn').simulate('click');
    expect(mockHandleChange).toBeCalledWith(fakeEvent.target);
  });

  it('should create task if input " " and click on submit', () => {
    const mockHandleChange = jest.fn();
    const wrapperComponent = shallow(
      <CreateTaskInput sendInputData={mockHandleChange} />
    );
    const fakeEvent = {
      target: { value: '' },
    };
    wrapperComponent.find('.create-task__input').simulate('change', fakeEvent);
    wrapperComponent.find('.btn').simulate('click');
    expect(mockHandleChange).toBeCalledWith({ value: '' });
  });

  it('should display this render', () => {
    const mockHandleChange = jest.fn();
    const wrapperComponent = shallow(
      <CreateTaskInput sendInputData={mockHandleChange} />
    );
    expect(wrapperComponent.find('.create-task')).toMatchSnapshot();
  });
});
