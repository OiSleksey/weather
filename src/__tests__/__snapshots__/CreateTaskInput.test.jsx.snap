// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateTaskInput.> should display this render 1`] = `
<div
  className="create-task"
>
  <input
    className="create-task__input"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="btn"
    onClick={[Function]}
  >
    Create
  </button>
</div>
`;
